internal/pkg/services/argus/utils/utils.go

```diff
@@ -56,6 +75,123 @@ func LoadPlanId(planName string, resp *argus.PlansResponse) (*string, error) {
 	}
 }
 
+func MapToUpdateScrapeConfigPayload(resp *argus.GetScrapeConfigResponse) (*argus.UpdateScrapeConfigPayload, error) {
+	if resp == nil || resp.Data == nil {
+		return nil, fmt.Errorf("no Argus scrape config provided")
+	}
+
+	data := resp.Data
+
+	basicAuth := mapBasicAuth(data.BasicAuth)
+	staticConfigs := mapStaticConfig(data.StaticConfigs)
+	tlsConfig := mapTlsConfig(data.TlsConfig)
+	metricsRelabelConfigs := mapMetricsRelabelConfig(data.MetricsRelabelConfigs)
+
+	var params *map[string]interface{}
+	if data.Params != nil {
+		params = utils.Ptr(mapParams(*data.Params))
+	}
+
+	payload := argus.UpdateScrapeConfigPayload{
+		BasicAuth:             basicAuth,
+		BearerToken:           data.BearerToken,
+		HonorLabels:           data.HonorLabels,
+		HonorTimeStamps:       data.HonorTimeStamps,
+		MetricsPath:           data.MetricsPath,
+		MetricsRelabelConfigs: metricsRelabelConfigs,
+		Params:                params,
+		SampleLimit:           utils.ConvertInt64PToFloat64P(data.SampleLimit),
+		Scheme:                data.Scheme,
+		ScrapeInterval:        data.ScrapeInterval,
+		ScrapeTimeout:         data.ScrapeTimeout,
+		StaticConfigs:         staticConfigs,
+		TlsConfig:             tlsConfig,
+	}
+
+	if payload == (argus.UpdateScrapeConfigPayload{}) {
+		return nil, fmt.Errorf("the provided Argus Scrape Sonfig payload is empty")
+	}
+
+	return &payload, nil
+}
+
+func mapMetricsRelabelConfig(metricsRelabelConfigs *[]argus.MetricsRelabelConfig) *[]argus.CreateScrapeConfigPayloadMetricsRelabelConfigsInner {
+	if metricsRelabelConfigs == nil {
+		return nil
+	}
+	mappedConfigs := make([]argus.CreateScrapeConfigPayloadMetricsRelabelConfigsInner, 0)
+	for _, config := range *metricsRelabelConfigs {
+		mappedConfig := argus.CreateScrapeConfigPayloadMetricsRelabelConfigsInner{
+			Action:       config.Action,
+			Modulus:      utils.ConvertInt64PToFloat64P(config.Modulus),
+			Regex:        config.Regex,
+			Replacement:  config.Replacement,
+			Separator:    config.Separator,
+			SourceLabels: config.SourceLabels,
+			TargetLabel:  config.TargetLabel,
+		}
+		mappedConfigs = append(mappedConfigs, mappedConfig)
+	}
+	return &mappedConfigs
+}
+
+func mapStaticConfig(staticConfigs *[]argus.StaticConfigs) *[]argus.UpdateScrapeConfigPayloadStaticConfigsInner {
+	if staticConfigs == nil {
+		return nil
+	}
+	mappedConfigs := make([]argus.UpdateScrapeConfigPayloadStaticConfigsInner, 0)
```

Comment:
Even with these new functions, I still think it would be more readable to have it `var mappedConfigs []argus.UpdateScrapeConfigPayloadStaticConfigsInner`, and it's the same thing